lib/tasks/db.rake

```diff
@@ -8,21 +8,36 @@ namespace :db do
   desc "Will update design document with what is on db/designs/*.coffee"
   task design: :environment do
     design_dir = Rails.root.join 'db', 'designs'
-    files = Dir.glob File.join design_dir, '**', '*.coffee'
+    coffee_files = Dir.glob File.join design_dir, '**', '*.coffee'
+    js_files = Dir.glob File.join design_dir, '**', '*.js'
+
     data = {}
 
+    files = coffee_files | js_files
+
     files.each do |filename|
       parts = filename[design_dir.to_s.length+1..-1].split '/'
       design_doc_name = parts.count == 1 ? Dolly::Document.design_doc : "_design/#{parts.first}"
 
-      name, key  = File.basename(filename).sub(/.coffee/i, '').split(/\./)
+      name, key  = File.basename(filename).sub(/.coffee|.js/i, '').split(/\./)
+
+      language = File.basename(filename) =~ /.js/ ? 'javascript' : 'coffeescript'
+
       key ||= 'map'
       source = File.read filename
 
-      vd = data[design_doc_name] ||= { 'views' => {}, 'filters' => {}, 'lists' => {}, 'lib' => {} }
+      vd = data[design_doc_name] ||= { 'views' => {}, 'filters' => {}, 'lists' => {}, 'lib' => {}, 'indexes' => {}}
```

Comment:
Mango `index` is different to the Lucene indexes, the later are set here. We will need something new for Mango indexes, as they are set with a `POST` request to `_index` interface